odxtools/audience.py

```diff
@@ -29,28 +58,20 @@ def from_et(et_element, doc_frags: List[OdxDocFragment]) \
         disabled_audience_refs = [OdxLinkRef.from_et(ref, doc_frags)
             for ref in et_element.iterfind("DISABLED-AUDIENCE-REFS/"
                                            "DISABLED-AUDIENCE-REF")]
-        is_supplier = et_element.get("IS-SUPPLIER", "true") == 'true'
-        is_development = et_element.get("IS-DEVELOPMENT", "true") == 'true'
-        is_manufacturing = et_element.get("IS-MANUFACTURING", "true") == 'true'
-        is_aftersales = et_element.get("IS-AFTERSALES", "true") == 'true'
-        is_aftermarket = et_element.get("IS-AFTERMARKET", "true") == 'true'
+        is_supplier = str_to_bool(et_element.get("IS-SUPPLIER"))
+        is_development = str_to_bool(et_element.get("IS-DEVELOPMENT"))
+        is_manufacturing = str_to_bool(et_element.get("IS-MANUFACTURING"))
+        is_aftersales = str_to_bool(et_element.get("IS-AFTERSALES"))
+        is_aftermarket = str_to_bool(et_element.get("IS-AFTERMARKET"))
 
         return Audience(enabled_audience_refs=enabled_audience_refs,
                         disabled_audience_refs=disabled_audience_refs,
-                        is_supplier=is_supplier,
-                        is_development=is_development,
-                        is_manufacturing=is_manufacturing,
-                        is_aftersales=is_aftersales,
-                        is_aftermarket=is_aftermarket)
-
+                        _is_supplier=is_supplier,
```

Comment:
don't add underscore to function arguments

Reply:
yeah, I also don't like it, but I did not have a better idea: the problem is that I want to keep the `.is_supplier` attribute for users, but we cannot directly use the value specified by the XML because if the attribute does not get specified, it ought to be interpreted as `True` whilst python interprets `None` as false. maybe use another prefix for the "raw" values for such cases? which one?

Reply:
I am saying don't add underscore to argument name in constructor, use underscore only for the private attribute. the rest of the logic is fine

Reply:
yeah, the issue is that I consider these attributes to be private (i.e., not meant to be used by "normal" users), but for dataclasses you need use the same name to pass them on construction time...

Reply:
you added logic to the class creation, they are no longer "data classes", thus needs to be converted to normal classes.

Reply:
while that's true in the strict sense, one can also look at these classes as primarily for storing data. (also, it is much more convenient and less error prone having to type these attributes only once...)

Reply:
having a function argument starts with underscore is unheard of in python, it makes it look like a private argument that's not supposed to be used publicly.

Reply:
alright. I changed this to using `_raw` suffixes. I don't really like this either, but I think it is not worse than using an underscore prefix...